packages/react-ui-core/src/Photo/Photo.js

```diff
@@ -0,0 +1,62 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import classnames from 'classnames'
+import themed from 'react-themed'
+
+@themed(['Photo'], {
+  pure: true,
+})
+
+export default class Photo extends PureComponent {
+  static propTypes = {
+    url: PropTypes.string,
+    fallbackUrl: PropTypes.string,
+    theme: PropTypes.object,
+    className: PropTypes.string,
+  }
+
+  static defaultProps = {
+    theme: {},
+  }
+
+  static getDerivedStateFromProps(props, state) {
+    if (!state.url || (state.url !== props.url)) {
+      return {
+        url: props.url,
+        error: false,
+      }
+    }
+
+    return null
+  }
+
+  fallback() {
+    this.setState({ error: true })
+  }
+
+  render() {
+    const {
+      theme,
+      className,
+      url: _,
+      fallbackUrl,
+      ...rest
+    } = this.props
+
+    const { url, error } = this.state
+
+    /* eslint-disable jsx-a11y/alt-text */
+    return (
+      <img
+        src={error ? fallbackUrl : url}
+        className={classnames(
+          className,
+          theme.Photo
+        )}
+        onError={!error && this.fallback}
```

Comment:
Nice, looks like this will not get caught in any type of a loop. And if you update the url, the new url can still have a fallback.

One question I have is around performance - since this is a stateful component, are there any concerns if we have, say, 100 of these components on the page?

Reply:
How else do you propose we do it? I guess we could try to remove state, but it might be hacky 🤷‍♂️

Reply:
I share Fitz's concerns about it being stateful in general, but it doesn't look like we're in danger of triggering any useless re-renders as far as I can tell.

Reply:
How about something like this instead: https://gist.github.com/tadjohnston/b7fa785b4e90f551f6a634908e5c3d35